akka-http-core/src/main/scala/akka/http/scaladsl/model/HttpMessage.scala

```diff
@@ -145,6 +147,16 @@ object HttpMessage {
       case HttpProtocols.`HTTP/1.1` ⇒ connectionHeader.isDefined && connectionHeader.get.hasClose
       case HttpProtocols.`HTTP/1.0` ⇒ connectionHeader.isEmpty || !connectionHeader.get.hasKeepAlive
     }
+
+  class DiscardedEntity(f: Future[Done]) extends akka.http.javadsl.model.HttpMessage.DiscardedEntity {
+    def future: Future[Done] = f
+    def completionStage: CompletionStage[Done] = FutureConverters.toJava(f)
+  }
+
+  implicit class HttpMessageDiscardEntity(val httpMessage: HttpMessage) extends AnyVal {
```

Comment:
Sorry, strike that I see that now we can make it work out well actually.
Please add a `final` to the class too.